.github/workflows/docs-gpu.yml

```diff
@@ -1,20 +1,29 @@
-name: Documentation
+name: Documentation of GPU
 
 on:
   [push, pull_request]
 
+defaults:
+  run:
+    # The slightly odd shell call is to force bash to read .bashrc, which is
+    # necessary for having conda behave sensibly.
+    shell: bash -l {0}
```

Comment:
A thought: your original travails with conda may have been to do with this line - I had forgotten, but to get conda to work correctly before, I had to force every subshell to be a login bash shell, so it would read the extra bits and bobs that conda adds to the `.profile` or whatever (rather than me manually hardcoding the environment paths).  If you're running in a container, I guess either `.profile` didn't exist, or maybe this shell variable wasn't getting passed down?

Reply:
Agreed. The self-hosted runner environment does seem a little different.